src/pickers.rs

```diff
@@ -15,6 +15,33 @@ pub trait Picker: std::fmt::Debug + Sync + Send {
     fn pick<'a>(&self, choices: &'a [Choice], scores: &Query<&Score>) -> Option<&'a Choice>;
 }
 
+/**
+Picker that chooses the `Choice` with the highest non-zero [`Score`], and the first highest in case of a tie.
+
+### Example
+
+```no_run
+Thinker::build()
+    .picker(HighestThenFirst)
+    // .when(...)
+```
+ */
+#[derive(Debug, Clone, Default)]
+pub struct HighestThenFirst;
+
+impl Picker for HighestThenFirst {
+    fn pick<'a>(&self, choices: &'a [Choice], scores: &Query<&Score>) -> Option<&'a Choice> {
+        let mut choices = choices
+            .iter()
+            .map(|v| (FloatOrd(v.calculate(scores)), v))
+            .filter(|v| v.0 .0 > 0.)
+            .collect::<Vec<(FloatOrd, &Choice)>>();
+        choices.sort_by_key(|v| v.0);
+        choices.reverse();
+        choices.first().map(|v| v.1)
```

Comment:
I'm worried about doing this much allocation in a picker. Perhaps a `.reduce` can be used to do this calculation in a single, non-allocating pass?